pkg/pull/pull.go

```diff
@@ -424,11 +424,19 @@ func Pull(upstreamURI string, pullOptions PullOptions) (string, error) {
 	// for writing Common Midstream, every chart and subchart is in this map as Helm Midstreams will be processed later in the code
 	commonWriteMidstreamOptions.UseHelmInstall = map[string]bool{}
 	for _, v := range newHelmCharts {
-		commonWriteMidstreamOptions.UseHelmInstall[v.Spec.Chart.Name] = v.Spec.UseHelmInstall
+		chartBaseName := v.Spec.Chart.Name
+		// the helmBase may have a chart name prefix removed - we must find the base name instead of the original chart name
```

Comment:
I'd rather do the reverse of this, and just not remove the prefix in question
but this will work, I suppose
just feels like more complexity